provider/pkg/provider/stack_tags.go

```diff
@@ -167,3 +176,12 @@ func (st *PulumiServiceStackTagResource) Read(req *pulumirpc.ReadRequest) (*pulu
 
 func (st *PulumiServiceStackTagResource) Configure(_ PulumiServiceConfig) {
 }
+
+func splitStackTagId(id string) (string, string, string, string, error) {
+	// format: organization/project/stack/tagName
+	s := strings.Split(id, "/")
+	if len(s) != 4 {
+		return "", "", "", "", fmt.Errorf("%q is invalid, must be in organization/project/stack/tagName fromat", id)
```

Comment:
NP: `format`*

also wouldn't be a bad idea to give names to the output types in the heading